### DISCUSSION TITLE
Anyway to represent a parent / child relasionship and query for it?

### DISCUSSION THREAD

**Discussion by TannerS** [sequelize/sequelize#17651] [Help & Questions]
I have a situations where I need table that has a column and Fk to the same table to represent a parent/child

so, take this example

Product table with a parent_id column that represents the id of a row in the same products table. I am not in anyway an expert at sequelize, but i was able to create the column and fk in a migration file, but in an attempt to find a way to query this table and have the output put the parents objects for a given object fails. I tried using aliases, I associated them in diff ways but no luck

is this possible? thanks!

**Comment by katelyngrimoldby** [comment]
Are you just looking for a way to get all of the children belonging to a given parent? If so, you can use special methods provided you explicitly define the one-to-many relationship within your model. There's more information [here](https://sequelize.org/docs/v6/core-concepts/assocs/#special-methodsmixins-added-to-instances) for v6.

For some detail on how sequelize would structure the data, the parent relationship is the foreign key on the children. The child relationship(s) don't get inserted into the database itself, instead sequelize stores that behind the scenes. So for example to get an array of all the children of one parent, you would need to use the getMany method sequelize offers on the parent (described in the link above). To get the parent of one child, you can either use the findByPk method using the child's foreign key, or if you declare your belongsTo in the model, use the getOne method sequelize offers on the children.

I hope this helps!
